`Development`: Comments and ratings for review page

MoSchmidt opened the pull request:
<!-- Thanks for contributing to TumApply! Before you submit your pull request, please make sure to check all tasks by putting an x in the [ ] (don't: [x ], [ x], do: [x]). Remove not applicable tasks and do not leave them unchecked -->
<!-- If your pull request is not ready for review yet, create a draft pull request! -->

### Checklist

#### General

<!-- Remove tasks that are not applicable for your PR. Please only put the PR into ready for review, if all relevant tasks are checked! -->
<!-- You only need to choose one of the first two check items: Generally, test on the test servers. -->
<!-- If it's only a small change, testing it locally is acceptable, and you may remove the first checkmark. If you are unsure, please test on the test servers. -->

- [x] I tested **all** changes and their related features with **all** corresponding user types.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://confluence.aet.cit.tum.de/spaces/AP/pages/257786006/Language+Guidelines+Client).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://confluence.aet.cit.tum.de/spaces/AP/pages/256311714/PR+Guidelines).

#### Client

- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data (e.g. using paging).
- [x] I **strictly** followed the principle of **data economy** for all client-server REST calls.
- [x] I **strictly** followed the [client coding and design guidelines](https://confluence.aet.cit.tum.de/spaces/AP/pages/256311716/Client+Guidelines).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://confluence.aet.cit.tum.de/spaces/AP/pages/256311925/Client+Test+Guidelines).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.


### Motivation and Context

This PR continues the refactoring of the review page to improve consistency, usability, and reviewer collaboration. While the first step focused on restructuring applicant details and documents, this second part introduces a unified design for ratings and internal comments. Reviewers can now provide structured Likert-scale ratings and collaborate via comments directly within the application detail view. Additionally, sections are visually enhanced with icons, and document handling has been improved with clearer tooltips and separation of shown vs. extra files. Together, these changes bring the review page closer to a polished, cohesive evaluation workflow.

### Description

#### Client

* **General UI**

  * `Section` components now support optional icons, making headings more recognizable.
  * Updates application detail layout to show **Ratings** and **Comments** sections alongside Documents.

* **Documents**

  * `DocumentSection` now splits displayed vs. extra documents and shows an info tooltip listing hidden ones.

* **Ratings**

  * Introduces `RatingSection` with:

    * Reviewer’s own rating (interactive Likert scale from -2 to +2).
    * Overview of ratings from other reviewers.
    * Automatic persistence of changes with error handling and UI rollback.
  * Refactors `RatingComponent` to use semantic Likert values instead of 1–5 indices.

* **Comments**

  * Introduces `CommentSection` for collaborative internal discussion:

    * Displays existing comments with edit/delete support.
    * Allows adding new comments with placeholder text.
    * Comments auto-reload when switching applications.
  * Refactors `Comment` component for:

### Steps for Testing

## Steps for Testing

- [x] **Application Detail Layout**
   - Open an application detail page → verify all sections (Applicant, Education, Application, Documents, Ratings, Comments) render with icons.

- [x]  **Documents**
   - Check the tooltip on “+N more documents” lists hidden documents.

- [x]  **Ratings**
   - Set your rating using the Likert scale (-2 to +2).
   - Refresh page → rating should persist.
   - Check sorting by “Rating (Worst to Best)” and “Rating (Best to Worst)” in the list view.

- [x]  **Comments**
   - Add a new comment → should appear immediately.
   - Edit an existing comment → changes should persist.
   - Delete a comment → it should disappear without reload.
   - Switch between applications → comments reset and reload correctly.


Prerequisites:

1. Log in as a professor
2. go to review page (details)


#### Code Review

- [x] Code Review 1

#### Manual Tests

- [x] Test 1

### Screenshots

<img width="3830" height="2071" alt="grafik" src="https://github.com/user-attachments/assets/ba449cc8-db8c-44e9-994f-5a784b3f54e2" />

MoSchmidt commented on src/main/webapp/i18n/de/global.json:
```diff
@@ -338,6 +338,9 @@
       "motivation": "Motivation",
       "skills": "Fähigkeiten",
       "researchExperience": "Forschungserfahrung"
+    },
```
I was not sure if we would add other comments in the future. As the placeholder would probably be the same, I put it into the global. But I can also move it to the evluation if you think this is a better place for it?evaluation

marc-fett replied:
I would either move it to evaluation or keep it in global or application if we reuse the same names in applications. At the moment, we have some of these texts duplicated.

MoSchmidt commented on src/main/webapp/app/shared/components/molecules/comment/comment.ts:
```diff
@@ -1,49 +1,74 @@
+    return !this.isCreate() && this.commentId() !== undefined && this.editingId() === this.commentId();
+  });
+
+  protected _updateDraftEffect = effect(() => {
```
No this can not be removed. onInput updates draft as the user types (and text in create mode). Without the effect, parent changes wouldn’t reach the textarea.

MoSchmidt commented on src/main/webapp/app/evaluation/components/comment-section/comment-section.html:
```diff
@@ -0,0 +1,36 @@
+          class="flex-1"
+          [commentId]="c.commentId!"
+          [author]="c.author ?? ''"
+          [createdAt]="c.createdAt ?? ''"
```
Adapted it like this

MoSchmidt commented on src/main/webapp/app/evaluation/application-detail/application-detail.component.html:
```diff
@@ -162,11 +162,21 @@ <h1 jhiTranslate="evaluation.reviewHeader">Applications - Review</h1>
-                [idsDTO]="currentDocumentIds()"
-                [applicationId]="currentApplication()?.applicationDetailDTO?.applicationId ?? ''"
-              />
+            <jhi-section title="evaluation.details.documentTitle" icon="folder">
```
Added tooltips for links

MoSchmidt commented:
> Thank you for the fixes. I added some answers to the old comments and added 2 new comments.
> 
> While testing I found these issues:
> 
>     * Sorting is not possible anymore but that is probably intended since we removed the filter bar, right?
> 
>     * You can rate approved and rejected applicants. That is not a big issue and no showstopper for this PR but we should add an issue for that since it is confusing for users.
> 
>     * Comment shows “2 hours ago” even if I just created the comment. This is also no showstopper but we should take a look at this soon since it is also confusing for the users.

Thanks for your review:
- Sorting was not touched with this PR and is already reworked anyway
- That you still can comment on accepted/rejected applications is a good point and will be fixed in the future
- The time difference is caused by the server. Probably it has a different time zone. The client has nothing to to with the timestamp

marc-fett reviewed (approved):
Thanks for the explanation and fixes.
